File: leetcode/easy/1342/1342.js
/**
 * Runtime: 80 ms, faster than 65.29%
 * Memory Usage: 38.8 MB, less than 26.72% 
 */
var numberOfSteps  = function(num) {
    let cnt = 0;
    
    while (num > 0) {
        if ((num & 1) == 0) num = num / 2;
        else num -= 1;
        ++cnt;
    }
    
    return cnt;
};

File: leetcode/easy/1342/1342.cpp
/**
* Runtime: 0 ms, faster than 100.00% 
* Memory Usage: 6.3 MB, less than 31.73%
*/
class Solution {
public:
    int numberOfSteps (int num) {
        int cnt = 0;
        while (num > 0) {
            if (num&1 == 1) --num;
            else num >>= 1;
            
            ++cnt;
        }
        return cnt;
    }
};
